tests/test_set_tools.py

```diff
@@ -0,0 +1,97 @@
+import math
+
+from fair.constraint import LinearConstraint
+from fair.feature import Course, Section, Slot, Weekday
+from fair.item import ScheduleItem
+from fair.set_tools import (
+    is_monotonic_non_decreasing,
+    is_mrf,
+    is_submodular,
+    nonnegative_rank_value,
+    rank_value_leq_cardinality,
+)
+from fair.valuation import ConstraintSatifactionValuation
+
+
+def length_marginal_increasing(vals):
+    """FAIL submodular, PASS monotone non-decreasing"""
+    return len(vals) ** 2
+
+
+def length_marginal_decreasing(vals):
+    """PASS submodular, PASS monotone non-decreasing"""
+    return math.sqrt(len(vals))
+
+
+def additive(vals):
+    """PASS submodular, PASS monotone non-decreasing"""
+    return sum(vals)
+
+
+def budget_additive(vals):
+    """PASS submodular, PASS monotone non-decreasing"""
+    return min(sum(vals), 3)
+
+
+def additive_divide_by_rank(vals):
+    """FAIL submodular, FAIL monotone non-decreasing"""
+    return sum(vals) / (len(vals) + 1)
+
+
+def test_mrf(course: Course, slot: Slot, weekday: Weekday, section: Section):
+    features = [course, slot, weekday, section]
+    schedule = [
+        ScheduleItem(features, ["250", (1, 2), ("Mon",), 1], 0),
+        ScheduleItem(features, ["250", (4, 5), ("Mon",), 2], 1),
+        ScheduleItem(features, ["301", (4, 5), ("Mon",), 1], 2),
+    ]
+    # is not MRF
+    A = [[1, 1, 0], [0, 1, 1]]
+    b = [1, 1]
+    constraint = LinearConstraint(A, b, 3)
+    valuation = ConstraintSatifactionValuation([constraint])
+    assert not is_mrf(schedule, valuation.value)
+
```

Comment:
Would it be easy to add a positive example that uses a valuation function constructed from `ConstraintSatifactionValuation`, like above? If so, I think it would be a nice addition.